x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -312,6 +313,22 @@ export const endpointMiddlewareFactory: ImmutableMiddlewareFactory<EndpointState
         });
       }
 
+      // call the activity log api
+      try {
+        const activityLog = await coreStart.http.get<EndpointAction[]>(
+          `/api/endpoint/action_log/${selectedAgent(getState())}`
```

Comment:
use the `const` that you defined with the route instead of hardcoding it.

there is a utility named `resolvePathVariables()` in:
https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/x-pack/plugins/security_solution/public/management/pages/trusted_apps/service/utils.ts#L8-L11

That you could use to replace the path tokens. That utility is currently in `trusted_apps`, so I would suggest you move it higher in the directory structure so that it implies that its not trusted apps specific. I would recommend moving it to `x-pack/plugins/security_solution/public/management/common/utils.ts`

Reply:
Sweet. 👍

Reply:
done 46c187d66aea1cfaeb1d48efb179167737b4c0b0 and da1c32d0440f9f3b8a5c611fcc8473ae15b8d778